app/controllers/SessionsCategoryController.scala

```diff
@@ -0,0 +1,256 @@
+
+  implicit val modelsCategoriesFormat: OFormat[ModelsCategoryInformation] = Json.format[ModelsCategoryInformation]
+
+  def renderCategoryPage: Action[AnyContent] = adminAction.async { implicit request =>
+    categoriesRepository
+      .getCategories
+      .map(category => Ok(views.html.category(category)))
+  }
+
+  def addPrimaryCategory(categoryName: String): Action[AnyContent] = superUserAction.async { implicit request =>
+    val cleanedCategoryName = categoryName.trim
+
+    if (cleanedCategoryName.isEmpty) {
+      Future.successful(BadRequest("Primary category cannot be empty"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          if (categories.exists(_.categoryName.toLowerCase == cleanedCategoryName.toLowerCase)) {
+            Future.successful(BadRequest("Primary category already exists"))
+          } else {
+            categoriesRepository
+              .insertCategory(cleanedCategoryName)
+              .map { result =>
+                if (result.ok) {
+                  Logger.info(s"Primary category was successfully added $categoryName")
+                  Ok("Primary category was successfully added")
+                } else {
+                  Logger.error(s"Something went wrong while adding primary category $categoryName")
+                  BadRequest("Primary category cannot be added due to some error")
+                }
+              }
+          }
+        }
+    }
+  }
+
+  def addSubCategory(categoryName: String, subCategory: String): Action[AnyContent] = adminAction.async { implicit request =>
+    val cleanedSubCategoryName = subCategory.trim
+
+    if (cleanedSubCategoryName.isEmpty) {
+      Future.successful(BadRequest("Subcategory cannot be empty"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          categories.find(_.categoryName == categoryName)
+            .fold {
+              Logger.info(s"No primary category found for category name $categoryName")
+              Future.successful(BadRequest("No primary category found."))
+            } { categoryInfo =>
+              val newSubCategory = categoryInfo.subCategory.find(_.toLowerCase == cleanedSubCategoryName.toLowerCase)
+              newSubCategory.fold {
+                val subCategoryInfo = CategoryInfo(categoryName, List(cleanedSubCategoryName), categoryInfo._id)
+                categoriesRepository
+                  .upsert(subCategoryInfo)
+                  .map { result =>
+                    if (result.ok) {
+                      Logger.info(s"Sub-category was successfully added $subCategory")
+                      Ok("Sub-category was successfully added")
+                    } else {
+                      Logger.error(s"Something went wrong while adding sub-category $subCategory")
+                      BadRequest("Unsuccessful sub-category added")
```

Comment:
Change "Unsuccessful sub-category added" to "Sub category cannot be added due to some error"